ipfabric/tools/table_diff.py

```diff
@@ -0,0 +1,201 @@
+    "__le__",
+    "__lt__",
+    "__module__",
+    "__ne__",
+    "__new__",
+    "__post_root_validators__",
+    "__pre_root_validators__",
+    "__pretty__",
+    "__private_attributes__",
+    "__reduce__",
+    "__reduce_ex__",
+    "__repr__",
+    "__repr_args__",
+    "__repr_name__",
+    "__repr_str__",
+    "__rich_repr__",
+    "__schema_cache__",
+    "__setattr__",
+    "__setstate__",
+    "__signature__",
+    "__sizeof__",
+    "__slots__",
+    "__str__",
+    "__subclasshook__",
+    "__try_update_forward_refs__",
+    "__validators__",
+    "_abc_impl",
+    "_calculate_keys",
+    "_copy_and_set_values",
+    "_decompose_class",
+    "_enforce_dict_if_root",
+    "_get_value",
+    "_init_private_attributes",
+    "_iter",
+    "schema_json",
+    "parse_file",
+    "parse_obj",
+    "parse_raw",
+    "from_orm",
+    "Config",
+]
+
+
+class IpfTableDiff(BaseModel):
+    client: Any
+
+    def _check_tech_table_attr(self, tech_table: [str, tuple[str, str]]):
+        """Takes a 'dotted' string representing location of tech table.
+        Example:
+            for arp table, pass the string 'addressing.arp_table'
+            # >>>ipf.technology.addressing.arp_table
+        Alternatively, takes a tuple of 2 strings representing location of tech table
+        Example:
+            for arp table, pass a tuple ('addressing', 'arp_table')
+        Args:
+            tech_table: string or tuple representing location of tech table
+
+        Returns:
+            IPFClient attribute for location of table data to be used in diff
+        """
+        valid_tables = [method for method in dir(self.client.technology) if method not in TECH_TABLE_METHODS_IGNORE]
+        if isinstance(tech_table, (tuple, str)):
+            if isinstance(tech_table, str):
+                tech_table = tuple(tech_table.split("."))
+            if isinstance(tech_table, tuple) and len(tech_table) != 2:
+                logger.error(f"unknown table {tech_table}")
+                logger.info(f"valid tables: \n {valid_tables}")
+                return None
+            first_attr = tech_table[0]
+            second_attr = tech_table[1]
+            tech_table_instance = getattr(self.client.technology, f"{first_attr}", None)
+            if tech_table_instance:
+                return_client = getattr(tech_table_instance, f"{second_attr}", None)
+                if not return_client:
+                    valid_tables = [method for method in dir(return_client) if method not in TECH_TABLE_METHODS_IGNORE]
+                    logger.error(f"unable to determine nested table {tech_table}")
+                    logger.info(f"valid tables: \n {valid_tables}")
+                return return_client
+        else:
+            logger.error(f"unable to determine nested table {tech_table}")
+            logger.info(f"valid tables: \n {valid_tables}")
+            logger.info(f"{HELP_STR}")
+
+    def _check_inv_table_attr(self, inv_table: str):
```

Comment:
Is this still needed?  Don't see it referenced anywhere

Reply:
I removed this file, sorry for the confusion, my bad :pensive: